destination.go

```diff
@@ -68,12 +72,25 @@ func (d *Destination) Configure(ctx context.Context, cfg map[string]string) erro
 }
 
 func (d *Destination) Open(ctx context.Context) error {
-	conn, err := grpc.DialContext(ctx,
-		d.config.URL,
-		grpc.WithContextDialer(d.dialer),
-		grpc.WithInsecure(), //nolint:staticcheck // todo: will use mTLS with connection
-		grpc.WithBlock(),
-	)
+	var conn *grpc.ClientConn
+	var err error
+	if d.config.RateLimit > 0 {
+		conn, err = grpc.DialContext(ctx,
+			d.config.URL,
+			grpc.WithContextDialer(d.dialer),
+			grpc.WithInsecure(), //nolint:staticcheck // todo: will use mTLS with connection
+			grpc.WithBlock(),
+			// limit bandwidth
+			bwgrpc.WithBandwidthLimitedContextDialer(bwlimit.Byte(d.config.RateLimit), bwlimit.Byte(d.config.RateLimit), d.dialer),
+		)
+	} else {
+		conn, err = grpc.DialContext(ctx,
+			d.config.URL,
+			grpc.WithContextDialer(d.dialer),
+			grpc.WithInsecure(), //nolint:staticcheck // todo: will use mTLS with connection
+			grpc.WithBlock(),
+		)
+	}
```

Comment:
Nitpick ahead. I'm thinking about this code in the following: specific options are always there (WithInsecure, WithContextDialer...). If rate limiting is enabled, we add it on top of that. If mTLS is enabled, we add it on top of that.

If mTLS is enabled, we need to add to both blocks for example. What might be a good option as well, it to have a slice of options, and do something like:
```
options := WithContextDialer, WithBlock // standard ones
if rateLimitingEnabled {
    options = append(options, WithRateLimiting)
}
if mTLS {
    options = append(options, WithMTLS)  
}
```

Reply:
oh I love this! I had the same idea at the beginning but tried to append it to the connection after it was created, which didn't work, but your way should! will try it out
thanks for the suggestions!

Reply:
You're welcome!